rfd/0133-connect-my-computer.md

```diff
@@ -0,0 +1,584 @@
+and immediately logs back in, the cache will still contain the now gone node. This is a rough edge
+we’re willing to accept in the MVP over adding more complexity to handle this specific scenario.
+
+### Availability of the feature
+
+The feature is supported only on macOS and Linux because the Teleport SSH agent does not work on
+Windows.
+
+Additionally, CMC is meant to be available only for Teleport Team plan users. To achieve this, the
+app fetches the `WebConfig` of the cluster and checks if the [`isUsageBasedBilling`](https://github.com/gravitational/teleport/blob/53b0118cff73fd5e9788a397d3b8a0aadc75d63a/web/packages/teleport/src/config.ts#L43-L44)
+flag is `true`. [The Web UI temporarily uses this flag](https://github.com/gravitational/teleport/blob/53b0118cff73fd5e9788a397d3b8a0aadc75d63a/web/packages/teleport/src/teleportContext.tsx#L66-L76)
+to show certain CTAs and intends to switch to using cluster features instead. However, for CMC this
+flag should be sufficient. CMC in itself is not going to be considered a cluster feature.
+
+### Security
+
+#### Access to the node
+
+The biggest concern is of course that Connect My Computer enables the user to grant access to their
+own machine as the system user that they regularly use, potentially exposing sensitive data. This is
+not a concern when connecting any other SSH node to the cluster, as usually those server nodes are
+understood as shared computers.
+
+Connect My Computer is targeted at hobbyists, people who just set up their Cloud cluster. The UI for
+Connect My Computer will not be accessible if the user doesn't have permission to create join
+tokens. As long as the administrators of more populated clusters follow the principle of least
+privilege, Connect My Computer won't be available to users who are not able to add new nodes to the
+cluster anyway. The UI of Connect My Computer should also make it absolutely clear as to what is
+going to happen. It won't be a single button the user can click by mistake.
+
+We did consider using some kind of an ephemeral user account for the agent. However, eventually we
+want to allow the user to share a link to a shell session on their device, ala
+[Teleconsole](https://github.com/gravitational/teleconsole). Think of a dev inviting a teammate to
+debug an issue with their dev setup. This feature would be substantially less useful if instead of
+sharing a session of your regular system user, you had to share a session from some ephemeral or
+otherwise restricted user account.
+
+From the point of view of the cluster, a node connected through Connect My Computer is just a
+regular node. There's no way to enforce that only a certain cluster user can access it. By
+default, [the role](#setting-up-roles) is added only to the user running Connect My Computer.
+However, anyone with the ability to change user roles will be able to add the role to any other
```

Comment:
Thanks for the feedback, all valid points. [I did have similar concerns](https://gravitational.slack.com/archives/C03FJA391M3/p1686840789399879?thread_ts=1684857042.076319&cid=C03FJA391M3).

With the MVP we're aiming at people who just created a Team plan account so that they're able to say "Oh cool, I just connected my computer to the cluster with a few clicks". Your feedback further synthesized that belief in me that if we'd like to take this feature any further, we'd need reinforce it with some of the ideas that you suggested.

For example, we talked about implementing Teleconsole-like behavior of sending an invite link to the cluster which then automatically joins a moderated session. At this point, there's more than just a single person using the cluster. So I feel like this would require putting additional restrictions in place.

I like the idea with extending the agent to require manual approval. I had the idea of modifying the agent so that connecting to it requires the password of the system user. Ultimately, I don't think we'll ever have a use case where we want to allow Alice to access Bob's computer without, say, a moderated session.

Reply:
64194a8 mentions the security concerns in [the What section of the RFD](https://github.com/gravitational/teleport/blob/rfd/0133-connect-my-computer/rfd/0133-connect-my-computer.md#what). 7b938f4 incorporates your feedback into [the Security section](https://github.com/gravitational/teleport/blob/rfd/0133-connect-my-computer/rfd/0133-connect-my-computer.md#security) and links to your ideas on further reinforcing the security in the future.

Let me know if there's anything else we should handle in that regard.